src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -285,47 +290,91 @@ protected override void StopProcessing()
 
         private void ProcessConnectionByTCPPort(string targetNameOrAddress)
         {
+            if(Repeat.IsPresent)
+            {
+                Count = int.MaxValue;
+            }
+            else if (!MyInvocation.BoundParameters.ContainsKey("Count"))
+            {
+                Count = 1;
+            }
+
             if (!TryResolveNameOrAddress(targetNameOrAddress, out _, out IPAddress? targetAddress))
             {
                 return;
             }
 
-            TcpClient client = new TcpClient();
+            int successfulConnections = 0;
 
-            try
+            for (var i = 1; i <= Count; i++)
             {
-                Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
-                string targetString = targetAddress.ToString();
-
-                for (var i = 1; i <= TimeoutSeconds; i++)
+                TcpPortStatus testResult = new TcpPortStatus(
+                    i,
+                    Source,
+                    targetNameOrAddress,
+                    targetAddress.ToString(),
+                    TcpPort,
+                    0,
+                    TcpPortStatus.TcpConnectionTestResult.New
+                );
+
+                using(TcpClient client = new TcpClient())
```

Comment:
You can avoid the need for the extra braces and indentation here if you use the newer statement form instead:

```csharp
// old form
using (var client = new TcpClient())
{
    // code using the client object here
}

// v.s. new form:
using var client = new TcpClient();
// code using the client object here (client will be cleaned up at the end of the current scope)
```

Docs here: https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#using-declarations

Reply:
This is neat! 

I'll definitely look at implementing this, cleans things up quite a bit!

Reply:
Updated! 😁Thanks for the tip!